core/src/main/java/org/elasticsearch/index/query/SpanContainingQueryBuilder.java

```diff
@@ -29,47 +35,93 @@
 public class SpanContainingQueryBuilder extends AbstractQueryBuilder<SpanContainingQueryBuilder> implements SpanQueryBuilder<SpanContainingQueryBuilder> {
 
     public static final String NAME = "span_containing";
-    private SpanQueryBuilder big;
-    private SpanQueryBuilder little;
+    private final SpanQueryBuilder big;
+    private final SpanQueryBuilder little;
     static final SpanContainingQueryBuilder PROTOTYPE = new SpanContainingQueryBuilder();
 
     /**
-     * Sets the little clause, it must be contained within {@code big} for a match.
+     * @param big the big clause, it must enclose {@code little} for a match.
+     * @param little the little clause, it must be contained within {@code big} for a match.
      */
-    public SpanContainingQueryBuilder little(SpanQueryBuilder clause) {
-        this.little = clause;
-        return this;
+    public SpanContainingQueryBuilder(SpanQueryBuilder big, SpanQueryBuilder little) {
+        this.little = Objects.requireNonNull(little);
+        this.big = Objects.requireNonNull(big);
     }
 
     /**
-     * Sets the big clause, it must enclose {@code little} for a match.
+     * only used for prototype
      */
-    public SpanContainingQueryBuilder big(SpanQueryBuilder clause) {
-        this.big = clause;
-        return this;
+    private SpanContainingQueryBuilder() {
+        this.little = null;
+        this.big = null;
+    }
+
+    /**
+     * @return the big clause, it must enclose {@code little} for a match.
+     */
+    public SpanQueryBuilder big() {
+        return this.big;
+    }
+
+    /**
+     * @return the little clause, it must be contained within {@code big} for a match.
+     */
+    public SpanQueryBuilder little() {
+        return this.little;
     }
 
     @Override
     protected void doXContent(XContentBuilder builder, Params params) throws IOException {
-        if (big == null) {
-            throw new IllegalArgumentException("Must specify big clause when building a span_containing query");
-        }
-        if (little == null) {
-            throw new IllegalArgumentException("Must specify little clause when building a span_containing query");
-        }
         builder.startObject(NAME);
-
         builder.field("big");
         big.toXContent(builder, params);
-
         builder.field("little");
         little.toXContent(builder, params);
-
         printBoostAndQueryName(builder);
         builder.endObject();
     }
 
     @Override
+    protected Query doToQuery(QueryParseContext parseContext) throws IOException {
+        Query innerBig = big.toQuery(parseContext);
+        assert innerBig instanceof SpanQuery;
```

Comment:
EmptyQueryBuilder can not be inner SpanQueryBuilder because of `instanceof` checks in parser, also can't be used here in constructor, so I think were good here? The whole nested SpanQueryBuilder hierarchy shouldn't suffer from this problem, in theory it would be possible to nest an empty `{ }` in them but that already would not be accepted by the current DSL I think.